cmd/backend.go

```diff
@@ -1,18 +1,124 @@
 package main
 
 import (
+	"flag"
+	"fmt"
 	"github.com/grafana/grafana-build/pipelines"
 	"github.com/urfave/cli/v2"
+	"strings"
 )
 
-var TestBackendUnit = &cli.Command{
-	Name:   "test",
-	Action: PipelineAction(pipelines.GrafanaBackendTests),
+type ChoiceFlag struct {
+	Name         string
+	Usage        string
+	Value        string
+	HasBeenSet   bool
+	Choices      []string
+	Aliases      []string
+	defaultValue string
+	Category     string
+	DefaultText  string
+	Required     bool
+	Hidden       bool
+	EnvVars      []string
+}
+
+func (f *ChoiceFlag) String() string {
+	return cli.FlagStringer(f)
+}
+
+func (f *ChoiceFlag) Apply(set *flag.FlagSet) error {
```

Comment:
Doesn't `Apply` also need to change `f.HasBeenSet` to true like in other implementations?

Reply:
I wasn't sure about this. If I'm reading this correctly, [that variable is only set](https://github.com/urfave/cli/blob/v2.25.1/flag_bool.go#L124) when the flag is set by file or environment variable. I didn't implement that functionality, so I wasn't sure what to do with the HasBeenSet variable.